internal/impl/javascript/vm.go

```diff
@@ -0,0 +1,244 @@
+package javascript
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/textproto"
+	"strings"
+
+	"github.com/benthosdev/benthos/v4/public/service"
+	"github.com/dop251/goja"
+	"github.com/dop251/goja_nodejs/console"
+	"github.com/dop251/goja_nodejs/require"
+)
+
+var (
+	requireRegistry *require.Registry
+)
+
+func init() {
+	createRequireRegistry()
+	// TODO: Implement registry live reloading in dev mode? E. g. if a JS file is modified.
+}
+
+func createRequireRegistry() {
+	// Initialize global registry. This is where modules (JS files) live. This enables easy code re-use.
+	// TODO: We need to set a registry root folder. Can we make this configurable somehow? Setting it to the working directory for now.
+	requireRegistry = require.NewRegistry(require.WithGlobalFolders(""))
```

Comment:
Hey @kabukky, I still need to put time aside to properly play around with this, but I think the major question I have is whether the registry here could be instanciated per-processor, and how bad the performance hit would be in that case?